x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/querydsl/agg/TopHitsAgg.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.querydsl.agg;
+
+import org.elasticsearch.search.aggregations.AggregationBuilder;
+import org.elasticsearch.search.sort.FieldSortBuilder;
+import org.elasticsearch.search.sort.SortBuilder;
+import org.elasticsearch.search.sort.SortOrder;
+import org.elasticsearch.xpack.sql.querydsl.container.Sort;
+import org.elasticsearch.xpack.sql.type.DataType;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import static org.elasticsearch.search.aggregations.AggregationBuilders.topHits;
+
+public class TopHitsAgg extends LeafAgg {
+
+    private final String sortField;
+    private final SortOrder sortOrder;
+    private final DataType fieldDataType;
+    private final DataType sortFieldDataType;
+
+
+    public TopHitsAgg(String id, String fieldName, DataType fieldDataType, String sortField,
+                      DataType sortFieldDataType, SortOrder sortOrder) {
+        super(id, fieldName);
+        this.sortField = sortField;
+        this.sortOrder = sortOrder;
+        this.fieldDataType = fieldDataType;
+        this.sortFieldDataType = sortFieldDataType;
+    }
+
+    @Override
+    AggregationBuilder toBuilder() {
+        // Sort missing values (NULLs) as last to get the first/last non-null value
+        List<SortBuilder<?>> sortBuilderList;
+        if (sortField != null) {
```

Comment:
The code here is redundant since the field ordering code is duplicated.
```
if (sortField!=null) {
  sort.add(fieldSortBuilder(sortField);
}

sort.add(fieldSortBuilder(fieldName());
```